### DISCUSSION TITLE
[@types/youtube-player] missing types on event stateChange

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [DefinitelyTyped/DefinitelyTyped#66589] [Issues with a @types package]
The type for the event of, `on("stateChange", event)` are incomplete. The definition mentions CustomEvent or an object with `data: number;`

But the event also responds with a `target`

Would be nice to have the types definition for that target as it contains information that is not on the player itself, for example `event.target.videoTitle`

[a comment by an automated account is not shown]

**Comment by jurca** [comment]
The definitions reflect the documentation: https://developers.google.com/youtube/iframe_api_reference#Events

The event does have a target, sure, and after some inspection through https://developers.google.com/youtube/youtube_player_demo, I can tell you that the `target` is the player itself.

However, the TypeScript's type for `CustomEvent` has only a type parameter for the `detail` property, so there is no straight-forward way of specifying the type of the `target`. `CustomEvent & {target: YouTubePlayer}` feels hacky - but I acknowledge that that is subjective.

What are your thoughts on this matter?

**Comment by MrAmericanMike** [comment]
Not an expert on Types or on how to go about it, and yes, the target is the player itself from what I have seen. With the detail that for some reason `videoTitle` is missing when using the youtube-player library, but not when using the API directly.
